### ISSUE TITLE
compiler is generating strange component.d.ts ambient types

### ISSUE THREAD

**Issue by Hotell** [ionic-team/stencil#279]
**Stencil version:**

```
 @stencil/core@0.7
```

**I'm submitting a:**

- [x] bug report
- [ ] feature request
- [ ] support request => Please do not submit support requests here, use one of these channels: https://forum.ionicframework.com/ or https://stencil-worldwide.slack.com

**Current behavior:**

Following code generates unexpected JSX namespace definitions with props that are non existent in that component -> those props are generated for all components btw, very strange
![image](https://user-images.githubusercontent.com/1223799/32408896-1c2788ce-c1a1-11e7-883e-55c18cab1d3f.png)

**Expected behavior:**

properly generate ambient type defs

**Steps to reproduce:**

**Related code:**

```tsx
@Component({
  tag: 'ole-box',
  styleUrl: 'box.sass',
  scoped: true
})
export class OleBox {
  render() {
    return (
      <div class="box">
        <article class="media">
          <div class="media-left">
            <slot name="left"/>
          </div>
          <div class="media-content">
            <div class="content">
              <slot/>
            </div>
          </div>
        </article>
      </div>
    )
  }
}
```

**Comment by jthoms1** [comment]
Currently mode and color are props a that are made globally available on all components.  I hope to change this in the future.  Originally stencil was going to be Ionic 4.  We refactored it into a tool and this is one of the remaining tasks.

**Comment by jthoms1** [comment]
Fixed with 40d48e8550176d3cd4308842c80c0338f7b00b34.
